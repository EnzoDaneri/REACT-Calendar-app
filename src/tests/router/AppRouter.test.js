import React from 'react';


import { mount, configure } from 'enzyme';
import { Provider } from 'react-redux';
import Adapter from 'enzyme-adapter-react-16';
configure({adapter: new Adapter()});
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';
import { AppRouter } from '../../router/AppRouter';



const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );


// store.dispatch = jest.fn();



    describe('Pruebas en <AppRouter/>', () => {
   
    test('Debe mostrar el espere... ', () => {
     const initState = {
            auth: {
                checking: true
            }
     };
     const store = mockStore( initState );


    const wrapper = mount(

     <Provider store={ store }> 
       <AppRouter/>
     </Provider>

); 

//  expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('h5').exists() ).toBe(true);

        
 });


 test('Debe mostrar la ruta pÃºblica', () => {
    const initState = {
           auth: {
               checking: false,
               uid: null
           }
    };
    const store = mockStore( initState );


   const wrapper = mount(

    <Provider store={ store }> 
      <AppRouter/>
    </Provider>

); 

   expect( wrapper ).toMatchSnapshot();
   expect( wrapper.find('.login-container').exists() ).toBe(true);

       
});
    


test('Debe mostrar la ruta privada', () => {
    const initState = {
           calendar: {
               events: []
           },
           ui: {
               modalOpen: false
           },

           auth: {
               checking: false,
               uid: '123',
               name: 'Juan Carlos'
           }
    };
    const store = mockStore( initState );


   const wrapper = mount(

    <Provider store={ store }> 
      <AppRouter/>
    </Provider>

); 

   expect( wrapper ).toMatchSnapshot();
   expect( wrapper.find('.calendar-screen').exists() ).toBe(true);

       
});
    

});
