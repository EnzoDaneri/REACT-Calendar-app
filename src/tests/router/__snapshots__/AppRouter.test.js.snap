// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter/> Debe mostrar la ruta privada 1`] = `ReactWrapper {}`;

exports[`Pruebas en <AppRouter/> Debe mostrar la ruta p√∫blica 1`] = `ReactWrapper {}`;
